src/pages/Facility/settings/organizations/FacilityOrganizationIndex.tsx

```diff
@@ -80,38 +87,225 @@ export default function FacilityOrganizationIndex({
+        >
+          <TableCell
+            className={`${
+              isTopLevel
+                ? "bg-white font-bold text-gray-900"
+                : "bg-white font-medium text-gray-900"
+            } flex justify-between lg:flex-row flex-col pl-[var(--indent)] flex-wrap gap-2`}
+          >
+            <div className="flex items-center">
+              {children.length > 0 ? (
+                <Button
+                  size={"icon"}
+                  variant={"link"}
+                  onClick={() => toggleRow(org.id)}
+                >
+                  {expandedRows[org.id] ? (
+                    <CareIcon icon="l-angle-down" className="h-5 w-5" />
+                  ) : (
+                    <CareIcon icon="l-angle-right" className="h-5 w-5" />
+                  )}
+                </Button>
+              ) : org.parent && Object.keys(org.parent).length > 0 ? (
+                <CareIcon
+                  icon="l-corner-down-right-alt"
+                  className="h-4 w-4 text-gray-400 ml-4"
+                />
+              ) : (
+                <Button
+                  size={"icon"}
+                  variant={"link"}
+                  onClick={() => toggleRow(org.id)}
+                  className="px-0"
+                >
+                  {expandedRows[org.id] ? (
+                    <CareIcon icon="l-angle-down" className="h-5 w-5" />
+                  ) : (
+                    <CareIcon icon="l-angle-right" className="h-5 w-5" />
+                  )}
+                </Button>
+              )}
+              {org.name}
+            </div>
+            {isTopLevel && (
+              <div className="flex justify-between gap-5">
+                {children.length > 0 ? (
+                  <>
+                    <Button
+                      variant="white"
+                      size={"sm"}
+                      onClick={toggleAllChildren}
+                    >
+                      <CareIcon
+                        icon={allExpanded ? "l-minus" : "l-plus"}
+                        className="h-4 w-4 sm:h-2 sm:w-2"
+                      />
+                      <span className="hidden sm:inline">
+                        {t(allExpanded ? "collapse_all" : "expand_all")}
+                      </span>
+                    </Button>
+                    <Button variant="white" size={"sm"} asChild>
+                      <Link
+                        href={`/departments/${org.id}`}
+                        className="text-gray-900  flex items-center"
+                      >
+                        <CareIcon icon="l-eye" className="h-4 w-4" />
+                        <span className="hidden sm:inline">
+                          {t("see_details")}
+                        </span>
+                      </Link>
+                    </Button>
+                  </>
+                ) : (
+                  <Button variant="white" size={"sm"} asChild>
```

Comment:
- You don't need to duplicate the code, display the expand all button for the check.
- Reduce button size for smaller devices
- Align the see details button on right even when expand all doesn't exist (justify-end should work).
- Add some spacing between icon (l-corner-down-right-alt) and the text.